recipes-nodes/config-parser/config-parser.bb

```diff
@@ -10,8 +10,23 @@ SRC_URI = "file://config_parser.sh \
 
 S = "${WORKDIR}"
 
-# Default value for the config URL
-CLOUD_INIT_CONFIG_URL ?= "http://169.254.169.254/metadata/instance/compute/userData?api-version=2021-01-01&format=text"
+python () {
+    # Check if CLOUD_INIT_CONFIG_URL is set in the environment or in local.conf
+    cloud_init_config_url = d.getVar('CLOUD_INIT_CONFIG_URL')
+    
+    if cloud_init_config_url is None:
+        # If not set, check the original environment
+        origenv = d.getVar("BB_ORIGENV", False)
```

Comment:
Hm, what's the `BB_ORIGENV`?

Reply:
when you want to use env variable of the calling host that runs bitbake command, it fetches them from the BB_ORIGENV.

Without this, the only possible way to pass them as env variable is through the bitbake ecosystem.  Such as, writing them in local.conf or in the recipes etc...
But in our case, we want to pass them when building as build arguments from docker